src/handlers/AIHandler.ts

```diff
@@ -0,0 +1,280 @@
+import {
+	IHttp,
+	IHttpResponse,
+} from '@rocket.chat/apps-engine/definition/accessors';
+import { IUser } from '@rocket.chat/apps-engine/definition/users';
+import { QuickRepliesApp } from '../../QuickRepliesApp';
+import { SettingEnum } from '../config/settings';
+import {
+	AIProviderEnum,
+	AIusagePreferenceEnum,
+	IPreference,
+} from '../definition/helper/userPreference';
+
+class AIHandler {
+	constructor(
+		private app: QuickRepliesApp,
+		private http: IHttp,
+		private userPreference: IPreference,
+	) {}
+
+	public async handleResponse(
+		user: IUser,
+		message: string,
+		prompt: string,
+	): Promise<string> {
+		let aiProvider: string;
+
+		if (
+			this.userPreference.AIusagePreference ===
+			AIusagePreferenceEnum.Personal
+		) {
+			aiProvider = this.userPreference.AIconfiguration.AIProvider;
+		} else {
+			aiProvider = await this.app
+				.getAccessors()
+				.environmentReader.getSettings()
+				.getValueById(SettingEnum.AI_PROVIDER_OPTOIN_ID);
+		}
+
+		switch (aiProvider) {
+			case AIProviderEnum.SelfHosted:
+			case SettingEnum.SELF_HOSTED_MODEL:
+				return this.handleSelfHostedModel(user, message, prompt);
+
+			case AIProviderEnum.OpenAI:
+			case SettingEnum.OPEN_AI:
+				return this.handleOpenAI(user, message, prompt);
+
+			case AIProviderEnum.Gemini:
+			case SettingEnum.GEMINI:
+				return this.handleGemini(user, message, prompt);
+
+			default:
+				const errorMsg =
+					this.userPreference.AIusagePreference ===
+					AIusagePreferenceEnum.Personal
+						? 'AI not set up. Please Check your configuration'
+						: 'AI not set up. Please contact your administrator';
+
+				this.app.getLogger().log(errorMsg);
+				return errorMsg;
+		}
+	}
+
+	private getPrompt(message: string, prompt: string): string {
+		return `Write a reply to this message: "${message}". Ensure the reply is simple. Use the following as a prompt or response reply: "${prompt}" and make sure you respond with a well-written message.`;
+	}
```

Comment:
Sure, maybe we can provide user a input option to modify the prompt and update it the way it want in the configure modal,  it would be useful if someone want to change the tones of suggestion they way he wanted.